src/main/java/games/strategy/triplea/ai/proAI/util/ProTransportUtils.java

```diff
@@ -232,7 +232,8 @@ public static int getUnusedCarrierCapacity(final PlayerID player, final Territor
 
   public static List<Unit> interleaveUnitsCarriersAndPlanes(final List<Unit> units,
       final int planesThatDontNeedToLand) {
-    if (!(Match.anyMatch(units, Matches.unitIsCarrier()) && Match.anyMatch(units, Matches.unitCanLandOnCarrier()))) {
+    if (units.stream().noneMatch(Matches.unitIsCarrier())
+        || units.stream().noneMatch(Matches.unitCanLandOnCarrier())) {
```

Comment:
Just verifying for my own benefit that this expression still evaluates to `true` when `units` is empty.